arcgis-ios-sdk-samples/Edit data/Add features with contingent values/ContingentValuesTableViewController.swift

```diff
@@ -0,0 +1,303 @@
+        // Set the minimum and maximum possible buffer sizes.
+        let minValue = bufferSizeGroupContingentValues[0].minValue as! Int
+        let maxValue = bufferSizeGroupContingentValues[0].maxValue as! Int
+        bufferSizes = Array(minValue...maxValue)
+        // Select the buffer size if it is the only option available.
+        if bufferSizes.count == 1 {
+            selectedBufferSize = bufferSizes[0]
+        }
+        // Update the picker view.
+        bufferSizePickerView.reloadAllComponents()
+    }
+    
+    func createFeature(with status: AGSCodedValue) {
+        // Get the contingent values definition from the feature table.
+        let contingentValuesDefinition = featureTable?.contingentValuesDefinition
+        // Load the contingent values definition.
+        contingentValuesDefinition?.load { [weak self] error in
+            guard let self = self else { return }
+            if let error = error {
+                self.presentAlert(error: error)
+            } else if let feature = self.featureTable?.createFeature() as? AGSArcGISFeature {
+                // Create a feature from the feature table and set the initial attribute.
+                feature.attributes["Status"] = status.code
+                self.feature = feature
+            }
+        }
+    }
+
+    /// Ensure that the selected values are a valid combination.
+    func validateContingency() {
+        guard let featureTable = featureTable, let feature = feature else { return }
+        // Validate the feature's contingencies.
+        let contingencyViolations = featureTable.validateContingencyConstraints(with: feature)
+        if contingencyViolations.isEmpty {
+            // If there are no contingency violations in the array, the feature is valid and ready to add to the feature table.
+            doneBarButtonItem.isEnabled = true
+        } else {
+            // Present an alert if there are contingency violations.
+            presentAlert(title: "", message: "Invalid contingent values")
+        }
+    }
+    
+    // MARK: UI Functions
+    
+    /// Update the cell's right detail.
+    func editRightDetail(cell: UITableViewCell, rightDetailText: String?) {
+        cell.detailTextLabel?.text = rightDetailText
+    }
+    
+    /// Reset the cell states according to which values have already been selected.
+    func resetCellStates() {
+        if selectedStatus == nil {
+            protectionCell.textLabel?.isEnabled = false
+            protectionCell.isUserInteractionEnabled = false
+        } else {
+            protectionCell.textLabel?.isEnabled = true
+            protectionCell.isUserInteractionEnabled = true
+        }
+        if selectedProtection == nil {
+            bufferSizeCell.textLabel?.isEnabled = false
+            bufferSizeCell.isUserInteractionEnabled = false
+        } else {
+            bufferSizeCell.textLabel?.isEnabled = true
+            bufferSizeCell.isUserInteractionEnabled = true
+        }
```

Comment:
Instead of enabling these based on what "step" the user is on, consider exposing the rows one at a time. So when the view first appears, only the status row is visible. When a status has been selected, the protection row becomes visible. And so on. That would be pretty easy to do in `tableView(_: numberOfRowsInSection:)`.

Reply:
Hm, I'm trying it out now but I finding that it may cause confusion, having to use "magic numbers" to determine how many rows to delete. Also, I'd have to reload the table view data after each selection. I'd like to keep it the way it is now. What're your thoughts?